Mlem/Views/Tabs/Posts/Components/Post/Components/Community View/Post Item/Components/Post Header.swift

```diff
@@ -0,0 +1,110 @@
+//
+//  Post Header.swift
+//  Mlem
+//
+//  Created by Eric Andrews on 2023-06-11.
+//
+
+import Foundation
+import SwiftUI
+import CachedAsyncImage
+
+struct PostHeader: View {
+    // parameters
+    var post: APIPostView
+    var account: SavedAccount
+    
+    // constants
+    private let communityIconSize: CGFloat = 32
+    private let defaultCommunityIconSize: CGFloat = 24 // a little smaller so it looks nice
+    
+    var body: some View {
+        HStack {
+            HStack(spacing: 4) {
+                // community avatar and name
+                NavigationLink(destination: CommunityView(account: account, community: post.community, feedType: .all)) {
+                    communityAvatar
+                        .frame(width: communityIconSize, height: communityIconSize)
+                        .clipShape(Circle())
+                        .overlay(Circle()
+                            .stroke(.secondary, lineWidth: 1))
+                    Text(post.community.name)
+                        .bold()
+                }
+                Text("by")
+                // poster
+                NavigationLink(destination: UserView(userID: post.creator.id, account: account)) {
+                    Text(post.creator.name)
+                        .italic()
+                        .if(post.creator.admin) { viewProxy in
+                            viewProxy
+                                .foregroundColor(.red)
+                        }
+                        .if(post.creator.botAccount) { viewProxy in
+                            viewProxy
+                                .foregroundColor(.indigo)
+                        }
+                        .if(post.creator.name == "lFenix") { viewProxy in
+                            viewProxy
+                                .foregroundColor(.yellow)
+                        }
+                }
+            }
+            
+            Spacer()
+            
+            if (post.post.featuredLocal) {
+                StickiedTag(compact: false)
+            }
+            
+            // ellipsis menu TODO: implement
+            Image(systemName: "ellipsis")
+                // circle background provides some tap leniency
+                .background(Circle()
+                    .frame(width: 16, height: 16)
+                    .foregroundColor(.white))
+                .foregroundColor(.black)
+                .contextMenu {
```

Comment:
because this is a `.contextMenu`, it's only responding to long presses, meaning on the main feed tapping it is taking me into the post, and then in the post I still need to long press... this doesn't seem like the correct implementation - a `...` menu should open on tap, right?

cc @buresdv